### PR TITLE
Test client reroute on DB2 with SSL

### PR DESCRIPTION
**Pull request by KyleAure**
Verify behavior for DB2 JCC driver using clientReroute and SSL

### CODE REVIEW

File: dev/com.ibm.ws.jdbc_fat_db2/publish/servers/com.ibm.ws.jdbc.fat.db2/server.xml [hunk 1]
```diff
@@ -108,6 +107,41 @@
     <jdbcDriver libraryRef="DB2Lib"/>
     <properties.db2.jcc databaseName="${DB2_DBNAME}" portNumber="${DB2_PORT}" user="${DB2_USER}" password="${DB2_PASS}" />
   </dataSource>
+  
+  <!-- 
+    Verify client re-route config
+  -->
+  <dataSource id="ds-client-reroute" jndiName="jdbc/ds-client-reroute">
+    <jdbcDriver libraryRef="DB2Lib"/>
+    <properties.db2.jcc serverName="fake.host.name" portNumber="1234"
```
Comment (mswatosh) [inline dev/com.ibm.ws.jdbc_fat_db2/publish/servers/com.ibm.ws.jdbc.fat.db2/server.xml]:
```suggestion
    <properties.db2.jcc serverName="host.name.invalid" portNumber="1234" 
```
`.invalid` is a reserved top level domain for things like this. `fake.host.name` could be a real domain name

File: dev/com.ibm.ws.jdbc_fat_db2/publish/servers/com.ibm.ws.jdbc.fat.db2/server.xml [hunk 2]
```diff
@@ -108,6 +107,41 @@
...
+    clientRerouteAlternateServerName="${DB2_HOSTNAME}" clientRerouteAlternatePortNumber="${DB2_PORT}"
+    maxRetriesForClientReroute="1" retryIntervalForClientReroute="15"/>
+  </dataSource>
+  
+  <!--
+    Verify client re-route with tls + cert
+  -->
+  <dataSource id="ds-client-reroute-cert" jndiName="jdbc/ds-client-reroute-cert">
+    <jdbcDriver libraryRef="DB2Lib"/>
+    <properties.db2.jcc serverName="fake.host.name" portNumber="1234"
```
Comment (mswatosh) [inline dev/com.ibm.ws.jdbc_fat_db2/publish/servers/com.ibm.ws.jdbc.fat.db2/server.xml]:
```suggestion
    <properties.db2.jcc serverName="host.name.invalid" portNumber="1234" 
```

File: dev/com.ibm.ws.jdbc_fat_db2/publish/servers/com.ibm.ws.jdbc.fat.db2/server.xml [hunk 3]
```diff
@@ -108,6 +107,41 @@
...
+    maxRetriesForClientReroute="1" retryIntervalForClientReroute="15"
+    sslConnection="true" sslVersion="TLSv1.2" sslCertLocation="${server.config.dir}/security/server.crt"/>
+  </dataSource>
+  
+  <!--
+    Verify client re-route with tls + wrong cert
+  -->
+  <dataSource id="ds-client-reroute-wrong-cert" jndiName="jdbc/ds-client-reroute-wrong-cert">
+    <jdbcDriver libraryRef="DB2Lib"/>
+    <properties.db2.jcc serverName="fake.host.name" portNumber="1234"
```
Comment (mswatosh) [inline dev/com.ibm.ws.jdbc_fat_db2/publish/servers/com.ibm.ws.jdbc.fat.db2/server.xml]:
```suggestion
    <properties.db2.jcc serverName="host.name.invalid" portNumber="1234" 
```

**Comment by KyleAure** [comment]
#libby - check headers
#build - personal build

**Review by mswatosh — APPROVED** [review]

**Pull request merged**